scvi/module/_vae.py

```diff
@@ -279,6 +302,10 @@ def _get_generative_input(self, tensors, inference_outputs):
             else None
         )
 
+        if self.batch_embedding is not None:
```

Comment:
Out of curiosity: Is the embedding optimized to be good for the encoder or the decoder?

Reply:
The embeddings are by default fed into the decoder only